stdlib/universe/join_test.go

```diff
@@ -720,6 +720,82 @@ func TestMergeJoin_Process(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "join with mismatched schemas with null in group key",
+			spec: &universe.MergeJoinProcedureSpec{
+				On:         []string{"_time"},
+				TableNames: tableNames,
+			},
+			data0: []*executetest.Table{
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value", Type: flux.TFloat},
+						{Label: "key", Type: flux.TString},
+					},
+					KeyCols: []string{"key"},
+					Data: [][]interface{}{
+						{execute.Time(1), 1.0, "foo"},
+						{execute.Time(2), 2.0, "foo"},
+					},
+				},
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value", Type: flux.TFloat},
+					},
+					KeyCols: []string{},
+					Data: [][]interface{}{
+						{execute.Time(1), 1.5},
+						{execute.Time(2), 2.5},
+					},
+				},
+			},
+			data1: []*executetest.Table{
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value", Type: flux.TFloat},
+						{Label: "key", Type: flux.TString},
+					},
+					KeyCols: []string{"key"},
+					Data: [][]interface{}{
+						{execute.Time(1), 10.0, nil},
+						{execute.Time(2), 20.0, nil},
+					},
+				},
+			},
+			want: []*executetest.Table{
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value_a", Type: flux.TFloat},
+						{Label: "_value_b", Type: flux.TFloat},
+						{Label: "key_a", Type: flux.TString},
+						{Label: "key_b", Type: flux.TString},
+					},
+					KeyCols: []string{"key_a", "key_b"},
+					Data: [][]interface{}{
+						{execute.Time(1), 1.0, 10.0, "foo", nil},
+						{execute.Time(2), 2.0, 20.0, "foo", nil},
+					},
+				},
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value_a", Type: flux.TFloat},
+						{Label: "_value_b", Type: flux.TFloat},
+						{Label: "key_a", Type: flux.TString},
+						{Label: "key_b", Type: flux.TString},
+					},
+					KeyCols: []string{"key_b"},
+					Data: [][]interface{}{
+						{execute.Time(1), 1.5, 10.0, nil, nil},
+						{execute.Time(2), 2.5, 20.0, nil, nil},
+					},
```

Comment:
Aargh, I find this `join` so hard to reason about. I like `experimental.join` so much better. I *think* this looks like what I would expect. But this answer doesn't seem consistent with the result in the acceptance test?

Reply:
I have replicated the go test in the acceptance test and with less col it's simple to follow through.